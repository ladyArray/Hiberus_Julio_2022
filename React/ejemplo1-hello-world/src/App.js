import logo from './logo.svg';
import './App.css';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <h1>Bienvenidos al curso de React !!</h1>
      </header>
    </div>
  );
}

export default App;
