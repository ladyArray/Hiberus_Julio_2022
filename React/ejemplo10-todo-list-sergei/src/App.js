import {useState} from 'react'
import TodoForm from './components/TodoForm';
import TodoList from './components/TodoList';

const App = () =>  {

  const [formValue, setFormValue] = useState('')
  const [todoList, setTodoList] = useState([])

  const handleChange = e => {
    e.preventDefault()
    setFormValue(e.target.value)
  }

  const handleSubmit = (e) => {
    e.preventDefault()
    setTodoList(todoList.concat(formValue))
    setFormValue('')
  }
  return (
    <div className="App">
      <TodoForm formValue={formValue} handleChange={handleChange} handleSubmit={handleSubmit}/>
      <TodoList todoList={todoList}/>
    </div>
  );
}

export default App;
