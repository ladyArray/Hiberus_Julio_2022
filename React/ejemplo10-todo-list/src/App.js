import logo from './logo.svg';
import './App.css';
import TodoForm from './components/TodoForm';
import TodoList from './components/TodoList';
import React from 'react';

class App extends React.Component {

  constructor(){
    super();

    this.state = {
      arrayDatos: []
    }
  }

  handleSubmit = e => {
    // Cancelar el submit
    e.preventDefault();

    // Con el valor recibido lo agregamos al estado
    let valor = Object.fromEntries(new FormData(e.target));
    this.setState(
      {arrayDatos: this.state.arrayDatos.concat(valor.tarea)}
    );
    //console.log(this.state.arrayDatos)
  }

  render(){
    return (
      <div>
        <TodoForm onSubmit={this.handleSubmit} />
        <TodoList coleccion={this.state.arrayDatos} />
      </div>
    );
  }
}

export default App;
