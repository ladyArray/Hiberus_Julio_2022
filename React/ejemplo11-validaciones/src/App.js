import './App.css';
import React from 'react';

class App extends React.Component {

  isInvalid = false;

  constructor(props){
    super();

    this.state = {
      form: {
        nombre:'',
        edad:''
      }
    }
  }

  handleChange = e => {
    this.setState({
      form: {
        ...this.state.form,
        [e.target.name]: e.target.value
      }
    })
  }

  validar = () => {
    let errores = {};

    // Validar Nombre
    if (!(this.state.form.nombre).match(/^[A-Z][a-z]+$/)){
      errores['nombre'] = 'El nombre solo puede tener letras y la primera mayuscula';
    }

    if (!this.state.form.nombre.length > 0){
      this.isInvalid = true;
      errores['nombre'] = 'El nombre no debe estar vacio';
    }


    // Validad Edad
    if (this.state.form.edad < 18){
      this.isInvalid = true;
      errores['edad'] = 'Debes tener minimo 18 aÃ±os';
    }

    this.setState(
      {errors: errores}
    );
  }

  render(){
    return (
      <div className="App">
        <form>
          <div>
            <label>Nombre:</label>
            <input type="text" name="nombre" onChange={this.handleChange} />
            <span style={{color:'red'}}>
              {this.isInvalid && this.state.errors.nombre}
            </span>
          </div>

          <div>
            <label>Edad:</label>
            <input type="number" name="edad" onChange={this.handleChange} />
            <span style={{color:'red'}}>
              {this.isInvalid && this.state.errors.edad}
            </span>
          </div>

          <input type="button" onClick={this.validar} value="validar" />
        </form>
      </div>
    );
  }
}

export default App;
