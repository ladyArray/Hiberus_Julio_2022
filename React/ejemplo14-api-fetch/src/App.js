import './App.css';
import Usuarios from './components/Usuarios';

function App() {
  return (
    <div className="App">
      <Usuarios />
    </div>
  );
}

export default App;
