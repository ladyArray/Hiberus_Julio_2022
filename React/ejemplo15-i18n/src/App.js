import React from 'react'; 
import './App.css';
import i18n from './i18n';
import { withTranslation } from 'react-i18next';

import aleman from './images/banderas/alemania.png';
import francia from './images/banderas/francia.png';
import ingles from './images/banderas/inglaterra.png';
import spain from './images/banderas/spain.png';
import tanzania from './images/banderas/tanzania.png';


function App({t}) {

  const changeLanguage = (lng) => {
    i18n.changeLanguage(lng);
  }

  return (
    <div className="App">
      <img src={aleman} onClick={() => changeLanguage('de')} />
      <img src={francia}  onClick={() => changeLanguage('fr')} />
      <img src={ingles} onClick={() => changeLanguage('en')} />
      <img src={spain} onClick={() => changeLanguage('es')} />
      <img src={tanzania} style={{width: '150px'}} onClick={() => changeLanguage('tz')} />
      <h1>{t('mensaje')}</h1>
    </div>
   
  )
}

export default withTranslation()(App);
