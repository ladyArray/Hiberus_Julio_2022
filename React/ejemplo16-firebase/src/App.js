import logo from './logo.svg';
import './App.css';
import MisAlumnos from './components/MisAlumnos';

function App() {
  return (
    <div className="App">
      <MisAlumnos />
    </div>
  );
}

export default App;
