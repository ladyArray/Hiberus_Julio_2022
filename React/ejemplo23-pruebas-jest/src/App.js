import './App.css';

function App() {
  return (
    <div className="App">
      <button>Pulsame</button>
    </div>
  );
}

export default App;
