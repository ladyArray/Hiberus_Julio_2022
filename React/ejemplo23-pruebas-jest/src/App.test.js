import { create } from 'react-test-renderer';
import App from './App';

test('probar boton', () => {
    let dom = create(<App />);
    expect(dom.toJSON()).toMatchSnapshot();
});
