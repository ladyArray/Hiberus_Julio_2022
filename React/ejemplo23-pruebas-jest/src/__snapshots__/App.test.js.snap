// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`probar boton 1`] = `
<div
  className="App"
>
  <button>
    Pulsame
  </button>
</div>
`;
